SConstruct

```diff
@@ -720,12 +654,89 @@ elif methods.using_clang(env):
                 "support C++17. Supported versions are Apple Clang 10 and later."
             )
             Exit(255)
+        if env["debug_paths_relative"] and not vanilla and cc_version_major < 12:
+            print("Apple Clang < 12 doesn't support -ffile-prefix-map, disabling `debug_paths_relative` option.")
+            env["debug_paths_relative"] = False
     elif cc_version_major < 6:
         print(
             "Detected Clang version older than 6, which does not fully support "
             "C++17. Supported versions are Clang 6 and later."
         )
         Exit(255)
+    if env["debug_paths_relative"] and cc_version_major < 10:
+        print("Clang < 10 doesn't support -ffile-prefix-map, disabling `debug_paths_relative` option.")
+        env["debug_paths_relative"] = False
```

Comment:
To avoid repetition I only check the `not vanilla` case in the macOS/iOS branch, which should already gracefully handle disabling the option if Apple Clang version is too old. So I think it _should_ be fine with this check re-running for vanilla Clang in the main scope.